Source/Parallelization/GuardCellManager.cpp

```
@@ -91,16 +92,20 @@ guardCellManager::Init (
     int ngJy = ngy_tmp;
     int ngJz = ngz_tmp;
 
-    // When calling the moving window (with one level of refinement),  we shift
-    // the fine grid by 2 cells ; therefore, we need at least 2 guard cells
-    // on level 1. This may not be necessary for level 0.
+    // When calling the moving window (with one level of refinement), we shift
+    // the fine grid by a number of cells equal to the ref_ratio in the moving
+    // window direction. This may not be necessary for level 0.
     if (do_moving_window) {
-        ngx = std::max(ngx,2);
-        ngy = std::max(ngy,2);
-        ngz = std::max(ngz,2);
-        ngJx = std::max(ngJx,2);
-        ngJy = std::max(ngJy,2);
-        ngJz = std::max(ngJz,2);
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(ref_ratios.size() == 1,
+            "The number of grow cells for the moving window currently assumes only 2 levels.");
+        int max_r = ref_ratios[0][moving_window_dir];
```

Just to make sure that I understand correctly: The `[0]` here means that we are looking at the refinement ratios, in all directions, between level `0` and level `1`, since we allow only 2 levels so far, correct?

---

Yes, that's right. The assumption of 2 levels max was already implicit here on `development`, so I added an assertion to fail hard here if that is violated.